### PR TITLE
Fix version check partially

### PR DESCRIPTION
**Pull request by qwerty287**
ref #2748 

- fix link to releases
- fix jq syntax
- support rc versions (separate json field)

### CODE REVIEW

File: web/src/compositions/useVersion.ts [hunk 1]
```diff
@@ -56,27 +59,31 @@ export function useVersion() {
...
   onMounted(async () => {
     const versionInfo = await fetchVersion();
 
-    let needsUpdate = false;
+    let latest;
     if (versionInfo) {
       if (usesNext) {
-        needsUpdate = versionInfo.next !== current;
+        latest = versionInfo.next;
+      } else if (current.includes('rc')) {
```
Comment (anbraten) [inline web/src/compositions/useVersion.ts:74]: I am checking using '-' or using the semver package most of the time as you could also release '1.2.3-beta.0'.

Maybe using the semver package is in general a nice idea as we could check that the upstream version is newer.

File: .woodpecker/docs.yml [hunk 2]
```diff
@@ -75,7 +75,8 @@ steps:
       - git config --global user.name "woodpecker-bot"
       - git clone --depth 1 --single-branch git@github.com:woodpecker-ci/woodpecker-ci.github.io.git /repo
       # update latest and next version
-      - if [ "$CI_PIPELINE_EVENT" == "tag" ] ; then jq '.latest = ${CI_COMMIT_TAG}' /repo/version.json > /repo/version.json.tmp && mv /repo/version.json.tmp /repo/version.json ; fi
+      - if [[ "$CI_PIPELINE_EVENT" == "tag" && "${CI_COMMIT_TAG}" != *"rc"* ]] ; then jq '.latest = "${CI_COMMIT_TAG}"' /repo/version.json > /repo/version.json.tmp && mv /repo/version.json.tmp /repo/version.json ; fi
+      - if [[ "$CI_PIPELINE_EVENT" == "tag" ]] ; then jq '.rc = "${CI_COMMIT_TAG}"' /repo/version.json > /repo/version.json.tmp && mv /repo/version.json.tmp /repo/version.json ; fi
```
Comment (anbraten) [inline .woodpecker/docs.yml:79]: So the rc channel will always be the latest release including rc and stable ones, right?

**Review by lafriks — APPROVED** [review]

File: web/src/compositions/useVersion.ts (hunk 1 above)
Comment (anbraten) [reply web/src/compositions/useVersion.ts:74]: So you would ignore the check / case that the user is running a newer version than the upstream config is "aware" of for now?

File: .woodpecker/docs.yml (hunk 2 above)
Comment (anbraten) [reply .woodpecker/docs.yml:79]: Just checked again and the added `"` should solve the current ci issue as well:

```
+ if [ "$CI_PIPELINE_EVENT" == "push" ] ; then jq '.next = "next-0fc428aa8e"' /repo/version.json > /repo/version.json.tmp && mv /repo/version.json.tmp /repo/version.json ; fi
+ rsync -r --exclude .git --exclude CNAME --exclude index.yaml --exclude README.md --exclude version.json --delete docs/build/ /repo
jq: error: Invalid numeric literal at EOF at line 1, column 4 (while parsing '.0.0') at <top-level>, line 1:
.latest = v2.0.0            
jq: error: syntax error, unexpected LITERAL, expecting $end (Unix shell quoting issues?) at <top-level>, line 1:
.latest = v2.0.0            
jq: 2 compile errors
```

File: web/src/compositions/useVersion.ts (hunk 1 above)
Comment (anbraten) [reply web/src/compositions/useVersion.ts:74]: fine

**Review by anbraten — APPROVED** [review]

**Pull request merged**